compiler/rustc_borrowck/src/type_check/liveness/trace.rs

```diff
@@ -599,12 +607,16 @@ impl<'tcx> LivenessContext<'_, '_, '_, 'tcx> {
                                 t,
                             )
                         }
-                    }));
-                if let Some(r) = outlives_bounds.next()
-                    && !r.is_late_bound()
-                    && outlives_bounds.all(|other_r| {
-                        other_r == r
-                    })
+                    }))
+                    .collect();
+                // If we find `'static`, then we know the alias doesn't capture *any* regions.
+                // Otherwise, all of the outlives regions should be equal -- if they're not,
+                // we don't really know how to proceed, so we continue recursing through the
+                // alias.
```

Comment:
in general we could pick the "maximum" one, but that's ok, static is a good choice

Reply:
Tangentially, could we start linting on _multiple_ `+ 'lt` bounds on `impl Trait`-aliases? It's always a mistake, one which beginners may make, or even people mislead by the rustc diagnostic which (in the single lifetime case) suggests `+ 'lt` instead of `+ Captures<'lt>`:

```rs
warning: multiple lifetime bounds
 --> src/lib.rs:5:41
  |
5 | fn foo<'a, 'b>(it: (&'a (), &'b ())) -> impl 'a + 'b + Trait {
  |                                              ^^^^^^^
help: introduce a new lifetime `'c` and use it in the `impl` type
5 | fn foo<'a, 'b, 'c>(it: (&'a (), &'b ())) -> impl 'c + Trait {
                 ++++                                ++
note: you will then need to bound `'c` w.r.t. `'a` and `'b` accordingly
```

Reply:
@danielhenrymantilla: We could lint such a case, which would help for the case of several *explicit* outlives choices. Consider outlives that are deduced from supertrait bounds tho:

```
trait Outlives<'a>: 'a {}
```

and some type like `impl Outlives<'a> + 'b`. We currently have to choose between `'a` and `'b` here too. I wouldn't want to make such a lint trigger here because it's possible that the user doesn't even have a choice over the outlives bounds that are a consequence of elaborating `Outlives`.

---

I'd rather it just be smart and understand how to fix a maximal region, but leave that for future work.

Reply:
@compiler-errors we do have a way to query for the maximum, or we did at one point, but we can discuss it separately.